public/app/features/explore/LogsContainer.tsx

```diff
@@ -90,12 +108,29 @@ const selectItemUIState = (itemState: ExploreItemState) => {
     dedupStrategy,
   };
 };
+
+const logsSelector = (state: ExploreItemState) => state.logsResult;
```

Comment:
All these selectors should be closer to the reducer (IMHO) and I'm not really convinced that we should have side effects in selectors. 
Couldn't all this be moved to the reducer and then you'll have to do this calculation for every action affected by this in the reducer right?

Reply:
The thing is that there are no side effects in the selector, only data processing. 

My thinking here is that deduped logs are function of `logsResults x dedupStrategy x hiddenLogsLevel` so there is no point in keeping them in store, if all the information required to calculate those is already there. Keeping them in store will introduce some more complexity to the reducers as well.I.e. we will have to dispatch dedup calc action on e.g. query run, while selector gives us this updates for free on input change.

Reply:
Ok I understand, I'm not sure that I agree though 😄 , I think the PR looks good!

Reply:
Sure. Any benefits of keeping this calculated data in redux that comes to your mind?

Reply:
It moves the processing away from the component. I'm more used to a pattern where the processing is done by the reducers and the components just connect to the processed state. But I think that is just my preference I don't think it really has any impact in this case.

Reply:
Well, I think it's my bad I have put selector in `LogsContainer` component. It definitely should belong to i.e. `explore/selectors.ts` as selectors are component agnostic and are used to retrieve data (derived or not) only.

Reply:
I think moving the selectors there is an excellent suggestion!

Reply:
Moved in dd19ec3 :)